src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -230,8 +230,24 @@ var ReactDOMInput = {
     // are not resetable nodes so this operation doesn't matter and actually
     // removes browser-default values (eg "Submit Query") when no value is
     // provided.
-    if (props.type !== 'submit' && props.type !== 'reset') {
-      node.value = node.value;
+
+    switch (props.type) {
+      case 'submit':
+      case 'reset':
+        break;
+      case 'date':
+      case 'datetime':
+      case 'datetime-local':
+      case 'month':
+      case 'time':
+      case 'week':
+        // this fixes the no-show issue on iOS Safari: https://github.com/facebook/react/issues/7233
+        node.value = '';
```

Comment:
What's the reason for this assignment since the next line reassigns it to defaultValue? Also, line 249 looks like a no-op as well.